src/java/za/co/mip/ablduck/ABLDuck.java

```diff
@@ -0,0 +1,345 @@
+
+            //Class tree
+            ClassDataObject cls = new ClassDataObject();
+            cls.name = js.name;
+            cls.ext = js.ext;
+            cls.icon = "icon-class";
+
+            if (js.meta.isPrivate != null && js.meta.isPrivate)
+                cls.isPrivate = true;
+
+            dataJSObject.classes.add(cls);
+
+            //Create search object
+            SearchDataObject search = new SearchDataObject();
+            search.name = js.shortname;
+            search.icon = "icon-class";
+            search.url = "#!/api/" + js.name;
+            search.sort = 1;
+            search.meta = js.meta;
+
+            dataJSObject.search.add(search);
+
+            for (MemberObject member : js.members) {
+                search = new SearchDataObject();
+                search.name = member.name;
+                search.icon = "icon-" + member.tagname;
+                search.url = "#!/api/" + member.owner + "-method-" + member.name;
+                search.sort = 3;
+                search.meta = member.meta;
+    
+                dataJSObject.search.add(search);
+            }
+            
+            //Hierarchy
+            HierarchyResult result = new HierarchyResult();
+            result = determineClassHierarchy(js, result);
+
+            Collections.reverse(result.hierarchy);
+            js.superclasses.addAll(result.hierarchy);
+
+            js.members.addAll(result.inheritedmembers);
+
+            //Subclasses
+            for (Map.Entry<String, SourceJSObject> subclass:jsObjects.entrySet()) {
+                SourceJSObject subc = subclass.getValue();
+                if(subc.ext.equals(js.name)) 
+                    js.subclasses.add(subc.name);
+            }
+
+        }
+
+        //Write class js files out
+        for (Map.Entry<String, SourceJSObject> j:jsObjects.entrySet()) {
+            SourceJSObject js = j.getValue();
+
+            //Generate html
+            js.html = html.getClassHtml(jsObjects, js);
+    
+            File outputFile = new File(this.destDirOutput, js.name + ".js");
+            try (FileWriter file = new FileWriter(outputFile.toString())) {
+                file.write("Ext.data.JsonP." + js.name.replace(".", "_") + "(" + gson.toJson(js) + ");");
+            } catch (IOException ex) {
+                throw new BuildException(ex);
+            }
+        }
+
+        File dataFile = new File(this.destDir, "data.js");
+        try (FileWriter file = new FileWriter(dataFile.toString())) {
+            file.write("Docs = {\"data\":" + gson.toJson(dataJSObject) + "}");
+        } catch (IOException ex) {
+            throw new BuildException(ex);
+        }
+        
+    }
+
+    private HierarchyResult determineClassHierarchy(SourceJSObject curClass, HierarchyResult result) {
+        
+        String inherits = curClass.ext;
+
+        if (!"".equals(inherits)) {
+            result.hierarchy.add(inherits);
+
+            curClass = jsObjects.get(inherits);

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "curClass". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1226)